source/guide_prosody.rst

```diff
@@ -0,0 +1,393 @@
+ [isabell@stardust ~]$
+
+Install runtime lua-dependencies
+--------------------------------
+
+The following dependencies_ (``luasocket``, ``luaexpat``, ``luafilesystem`` and ``luasec``) are required:
+
+::
+
+ [isabell@stardust ~]$ luarocks install luasocket --local
+ luasocket [...] is now built and installed in [...]
+ [isabell@stardust ~]$ luarocks install luaexpat --local
+ luaexpat [...] is now built and installed in [...]
+ [isabell@stardust ~]$ luarocks install luafilesystem --local
+ luafilesystem [...] is now built and installed in [...]
+ [isabell@stardust ~]$ luarocks install luasec --local
+ luasec [...] is now built and installed in [...]
+ [isabell@stardust ~]$
+
+Further optional ones (``luadbi-mysql``, ``luabitop`` and ``luaevent``) can be installed with these commands:
+
+::
+
+ [isabell@stardust ~]$ luarocks install luadbi-mysql --local MYSQL_BINDIR="/usr/bin" MYSQL_INCDIR="/usr/include/mysql" MYSQL_LIBDIR="/usr/lib64"
+ luadbi [...] is now built and installed in [...]
+ luadbi-mysql [...] is now built and installed in [...]
+ [isabell@stardust ~]$ luarocks install luabitop --local
+ luabitop [...] is now built and installed in [...]
+ [isabell@stardust ~]$ luarocks install luaevent --local
+ luaevent [...] is now built and installed in [...]
+ [isabell@stardust ~]$
+
+.. note:: The variables ``*_BINDIR`` ``*__INCDIR`` and ``*_LIBDIR`` are necessary for correct linking the associated library because CentOS uses a different layout for those files than luarocks expects!
+
+To list the installed packages with their versions use the command ``luarocks list``.
+
+MySQL
+-----
+
+.. include:: includes/my-print-defaults.rst
+
+Create a database for prosody:
+
+::
+
+ [isabell@stardust ~]$ mysql -e "CREATE DATABASE ${USER}_prosody"
+ [isabell@stardust ~]$
+
+Installation
+============
+
+.. note:: Check out the latest stable release on https://prosody.im/downloads/source/.
+
+Download the latest stable release from source to ``~/prosody/prosody-0.11.5``:
+
+::
+
+ [isabell@stardust ~]$ mkdir ~/home/prosody
+ [isabell@stardust ~]$ cd ~/home/prosody
+ [isabell@stardust ~]$ wget https://prosody.im/downloads/source/prosody-0.11.5.tar.gz
+ [...]
+ [isabell@stardust ~]$ tar -xzf prosody-0.11.5.tar.gz
+
+Configure, build and install prosody:
+
+::
+
+ [isabell@stardust ~]$ cd ~/prosody/prosody-0.11.5
+ [isabell@stardust ~]$ ./configure --ostype=linux --prefix=$HOME --libdir=$HOME/prosody --datadir=$HOME/prosody --with-lua-include=/usr/include
+ Lua version detected: 5.1
+ Lua interpreter found: /usr/bin/lua...
+ Checking Lua includes... lua.h found in /usr/include/lua.h
+ Checking if Lua header version matches that of the interpreter... yes
+ Writing configuration...
+
+ Installation prefix: /home/isabell
+ Prosody configuration directory: /home/isabell/etc/prosody
+ Using Lua from: /usr
+
+ Done. You can now run 'make' to build.
+
+ [isabell@stardust ~]$ make
+ [...] 
+ [isabell@stardust ~]$ make install
+ [...] 
+ [isabell@stardust ~]$ 
```

Comment:
Its a nice aspect and I didnt now that you can run configure and make with the directory declaration. And thats maybe the problem. I think to configure, make and install the package IN the directory is the most known way to do this and for new users its a little bit confuse maybe. Other guides "make" it in the directory too ;) I want to keep this pattern.

Reply:
That is why I actually just mentioned it and didn't put it into the code-suggestion. It's just a personal taste of doing stuff because I also write shell scripts where `cd` is a very confusing element.
It is just an idea so people could also put the elements of your guide into a script, hit `enter` - and it's done :wink:
I know most guides do it the "cd-way", so another way could also be confusing. So just keep it simple. I agree.